src/VecSim/algorithms/hnsw/hnsw_factory.h

```diff
@@ -16,6 +16,8 @@
 namespace HNSWFactory {
 
 VecSimIndex *NewIndex(const HNSWParams *params, std::shared_ptr<VecSimAllocator> allocator);
+VecSimIndex *NewTieredIndex(const TieredHNSWParams *params,
```

Comment:
who calls this function right now other than the tests?

Reply:
Currently no one. In the future, it should be called from `VecSimIndex_New`. Should it be part of this PR?